File: src/main/java/com/task/interview/colliers/digital/repository/TransactionRepository.java
package com.task.interview.colliers.digital.repository;

import com.task.interview.colliers.digital.document.Transaction;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface TransactionRepository extends MongoRepository<Transaction, Long> {

    List<Transaction> findAllByAccountTypeIdIn(List<Long> accountTypeIds);

    List<Transaction> findAllByCustomerIdIn(List<Long> customerIds);

    List<Transaction> findAllByAccountTypeIdInAndCustomerIdIn(List<Long> accountTypeIds, List<Long> customerIds);

}

File: src/main/java/com/task/interview/colliers/digital/repository/TransactionRepository.kt
package com.task.interview.colliers.digital.repository

import com.task.interview.colliers.digital.document.Transaction
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface TransactionRepository : MongoRepository<Transaction?, Long?> {

    fun findAllByAccountTypeIdIn(accountTypeIds: List<Long?>?): List<Transaction?>?

    fun findAllByCustomerIdIn(customerIds: List<Long?>?): List<Transaction?>?

    fun findAllByAccountTypeIdInAndCustomerIdIn(accountTypeIds: List<Long?>?, customerIds: List<Long?>?): List<Transaction?>?

}